internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -202,6 +214,31 @@ func needsProxyResolver(state *resolver.State) bool {
 	return false
 }
 
+func parseTarget(target string) (string, error) {
+	if target == "" {
+		return "", fmt.Errorf("missing address")
+	}
+	if host, port, err := net.SplitHostPort(target); err == nil {
+		if port == "" {
```

Comment:
Based on the documentation of `net.SplitHostPort`, the `port` seems to be mandatory and it will return a `missing port in address` error in that case. So, I don't think we could have `err == nil` and `port == ""`. 

```
// SplitHostPort splits a network address of the form "host:port",
// "host%zone:port", "[host]:port" or "[host%zone]:port" into host or
// host%zone and port.
```

Please correct me if I'm wrong.